File: backend/src/main/java/io/fingersonbuzzers/backend/validation/ValidationResult.java
package io.fingersonbuzzers.backend.validation;

import java.util.HashSet;
import java.util.Set;

public record ValidationResult(Set<FieldError> fieldErrors) {

  public ValidationResult() {
    this(new HashSet<>());
  }

  public void reject(String fieldName, String errorCode) {
    fieldErrors.add(new FieldError(fieldName, errorCode));
  }

  public boolean hasErrors() {
    return !fieldErrors.isEmpty();
  }
}

File: frontend/src/helpers/validation-result.ts
export interface ValidationResult {
  fieldErrors: [FieldError]
}

export interface FieldError {
  fieldName: string,
  errorCode: string
}

export function findFieldError(validationResult: ValidationResult|undefined,
                               fieldName: string) {
  return validationResult?.fieldErrors.find(
    fieldError => fieldError.fieldName === fieldName
  );
}
